string[] status_flag_right
string[] status_flag_left
string[] fault_flag